src/main/java/com/robinpowered/sdk/RobinApi.java

```diff
@@ -134,6 +135,16 @@ public boolean hasCredential() {
     }
 
     /**
+     * Returns whether a the interface is authenticated with a valid credential.
+     * Valid credentials have non-null values.
+     *
+     * @return Whether a valid credential has been set.
+     */
+    public boolean isAuthenticated() {
+        return credential != null && credential.getValue() != null;
```

Comment:
`isAuthenticated` is a confusing name here, because it implies the credential is valid. 

What if we overload `hasCredential` with `public boolean hasCredential(boolean requireNonNullValue)`

or something to that affect.